docs/modules/documentation/containers/calendar/calendar-docs.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'app-calendar',
+    templateUrl: './calendar-docs.component.html'
+})
+export class CalendarDocsComponent implements OnInit {
+    selectedDate: string;
+    selectedFirstDate: string;
+    selectedLastDate: string;
+
+    updateDatePickerInputHandler(d) {
+        if (d.selectedDay.id !== 0) {
+            this.selectedDate = d.selectedDay.date.toDateString();
+        }
+
+        if (d.selectedFirstDay.id !== 0) {
+            this.selectedFirstDate = d.selectedFirstDay.date.toDateString();
+            this.selectedLastDate = d.selectedLastDay.date.toDateString();
+        }
+    }
+
+    myDisableFunction = function(d: Date): boolean {
+        let day = d.getDay();
+        return day === 6 || day === 0;
+    };
+
+    // Block days before/after any day
+    myBlockFunction = function(d: Date): boolean {
+        let firstDay = new Date(2018, 7, 25);
+        let lastDay = new Date(2018, 7, 30);
+        return d.getTime() > firstDay.getTime() && d.getTime() < lastDay.getTime();
+    };
+
+    calendarSingleHtml = `<fd-calendar [datePickerType]="'single'" [disableFunction]="myDisableFunction" [blockFunction]="myBlockFunction"></fd-calendar>
+    `;
+
+    exampleFunctionsHtml = `
+    Example Disable and Block Functions: 
```

Comment:
These code blocks do not seem to be code highlighted properly, maybe because of this plain text here?